hack/common.sh

```diff
@@ -1066,6 +1066,16 @@ function os::build::environment::withsource() {
 
   docker start "${container}" > /dev/null
   docker logs -f "${container}"
+
+  # extract content from the image
+  if [[ -n "${OS_BUILD_ENV_PRESERVE}" ]]; then
+    local workingdir="$(docker inspect -f '{{ index . "Config" "WorkingDir" }}' "${container}")"
+    IFS=: eval 'for path in ${OS_BUILD_ENV_PRESERVE}; do
```

Comment:
I would technically have to trap it (Otherwise trap could have different
IFS behavior).  That's partially why I don't like it.

On Tue, Aug 16, 2016 at 1:41 PM, Steve Kuznetsov <notifications@github.com>
wrote:

> In hack/common.sh
> <https://github.com/openshift/origin/pull/10458#discussion_r74983394>:
>
> > @@ -1066,6 +1066,16 @@ function os::build::environment::withsource() {
> >
> >    docker start "${container}" > /dev/null
> >    docker logs -f "${container}"
> > +
> > +  # extract content from the image
> > +  if [[ -n "${OS_BUILD_ENV_PRESERVE}" ]]; then
> > +    local workingdir="$(docker inspect -f '{{ index . "Config" "WorkingDir" }}' "${container}")"
> > +    IFS=: eval 'for path in ${OS_BUILD_ENV_PRESERVE}; do
>
> I guess this would leak to the rest of the function after this call, so if
> you care about that case:
>
> local OLD_IFS="${IFS}"
> IFS=:
> // for loopexport IFS="${OLD_IFS}"
>
> In either case, I think staying away from eval is best since it's a gold
> mine of gotchas and hidden behavior.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10458/files/4b046250753fe0e53bf15c100edb397d81ac6315#r74983394>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p1MCTVbvg4bhOH6eWa4kl-msXGXFks5qgfZbgaJpZM4Jlmyi>
> .
>